File: app/src/main/java/com/commit451/gitlab/viewHolder/LabelViewHolder.java
package com.commit451.gitlab.viewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.commit451.gitlab.R;
import com.commit451.gitlab.model.api.Label;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Label
 */
public class LabelViewHolder extends RecyclerView.ViewHolder {

    public static LabelViewHolder inflate(ViewGroup parent) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_label, parent, false);
        return new LabelViewHolder(view);
    }

    @BindView(R.id.title)
    public TextView textTitle;
    @BindView(R.id.color)
    public View viewColor;

    public LabelViewHolder(View view) {
        super(view);
        ButterKnife.bind(this, view);
    }

    public void bind(Label label) {
        textTitle.setText(label.getName());
        viewColor.setBackgroundColor(label.getColor());
    }
}

File: app/src/main/java/com/commit451/gitlab/viewHolder/LabelViewHolder.kt
package com.commit451.gitlab.viewHolder

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.commit451.addendum.recyclerview.bindView
import com.commit451.gitlab.R
import com.commit451.gitlab.extension.getColor
import com.commit451.gitlab.extension.getTitleColor
import com.commit451.gitlab.model.api.Label

/**
 * Label
 */
class LabelViewHolder(view: View) : RecyclerView.ViewHolder(view) {

    companion object {

        fun inflate(parent: ViewGroup): LabelViewHolder {
            val view = LayoutInflater.from(parent.context)
                    .inflate(R.layout.item_label, parent, false)
            return LabelViewHolder(view)
        }
    }

    private val textTitle: TextView by bindView(R.id.title)
    private val viewColor: View by bindView(R.id.color)

    fun bind(label: Label) {
        textTitle.text = label.name
        textTitle.setTextColor(label.getTitleColor())
        viewColor.setBackgroundColor(label.getColor())
    }
}
